### PR TITLE
gdb: explore stack of non-current fiber

### LINKED ISSUE #7733: GDB tooling: explore non-current fiber
- A way to look into a non-current fiber.
  - List all fibers of given thread (may be called `info fibers` similarly to `info threads`, alternative: `tt-cord`).
  - 'Switch' to another fiber to use `bt`/`frame`/`list` and so on (may be called `fiber N` or `tt-fiber N` similarly to `thread N`).
  - If such 'switch' is hard to implement, provide some analogues for `bt`/`frame`/`list` that works for given fiber.

### PR DESCRIPTION
**Pull request by elhimov**
This commit introduce commands:
- info tt-fibers (Display fibers of the current cord)
- tt-fiber FID (Select fiber which stack need to be explored)

After switching to the desired fiber its stack can be explored with
general stack commands of GDB (bt, frame, etc.)

Closes #7733

NO_DOC=gdb extension
NO_CHANGELOG=gdb extension
NO_TEST=gdb extension

### CODE REVIEW

File: tools/tarantool-gdb.py [hunk 1]
```diff
@@ -2555,3 +2556,210 @@ def invoke(self, arg, from_tty):
...
+    def __call__(self, pending_frame):
+        if self.fiber() == fiber():
+            FiberUnwinderFrameFilter.set_skip_frame_sp(None)
+            return None
+
+        if pending_frame.level() > 0:
+            return None
+
+        orig_rsp = pending_frame.read_register('rsp')
+        orig_rip = pending_frame.read_register('rip')
```
[the remark this answers is not in the document]
Comment (elhimov) [reply tools/tarantool-gdb.py]: Fixed

File: tools/tarantool-gdb.py [hunk 2]
```diff
@@ -2555,3 +2556,210 @@ def invoke(self, arg, from_tty):
...
+        self.__fibers = {}
+        gdb.unwinder.register_unwinder(None, self, True)
+
+    def __call__(self, pending_frame):
+        if self.fiber() == fiber():
+            FiberUnwinderFrameFilter.set_skip_frame_sp(None)
+            return None
+
+        if pending_frame.level() > 0:
+            return None
```
[the remark this answers is not in the document]
Comment (elhimov) [reply tools/tarantool-gdb.py:2685]: Thank you for details. Fixed

[the remark this answers is not in the document]
Comment (locker) [reply tools/tarantool-gdb.py:2685]: Now it crashes. Is it a bug in my gdb version?

<details>

```
(gdb) info tt-fiber
  Id     Target   Name                               Stack              Function
  112    TtFiber  "metrics_collector"                0x7fb8dee7fc68     0x563db468d2c1 <lua_fiber_run_f>
  111    TtFiber  "feedback_daemon"                  0x7fb8dedffbf8     0x563db468d2c1 <lua_fiber_run_f>
  110    TtFiber  "guard of feedback_daemon"         0x7fb8ded7fc68     0x563db468d2c1 <lua_fiber_run_f>
  108    TtFiber  "vinyl.vylog_flusher"              0x7fb8dec7fda8     0x563db44f9ef5 <vy_log_flusher_f>
  107    TtFiber  "vinyl.scheduler"                  0x7fb8f73ffd58     0x563db450c126 <vy_scheduler_f>
  106    TtFiber  "memtx.gc"                         0x7fb8f737fdf8     0x563db448d389 <memtx_engine_gc_f(typedef __va_list_tag __va_list_tag *)>
  105    TtFiber  "checkpoint_daemon"                0x7fb8f72ffd28     0x563db456f6e8 <gc_checkpoint_fiber_f>
  104    TtFiber  "gc"                               0x7fb8f727fdd8     0x563db456e9ee <gc_cleanup_fiber_f>
  103    TtFiber  "interactive"                      0x7fb8f71ffb58     0x563db468a2c5 <run_script_f>
  102    TtFiber  "on_shutdown"                      0x7fb8f717ff98     0x563db4433d57 <on_shutdown_f(typedef __va_list_tag __va_list_tag *)>
  101    TtFiber  "lua"                              0x7fb8f70ffd28     0x563db468d2c1 <lua_fiber_run_f>
* 1      TtFiber  "sched"                          * 0x7ffd24c56208     0x0
(gdb) tt-fiber 102
/build/gdb-ktlO15/gdb-9.2/gdb/regcache.c:166: internal-error: int register_size(gdbarch*, int): Assertion `regnum >= 0 && regnum < gdbarch_num_cooked_regs (gdbarch)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) [answered Y; input not from terminal]

This is a bug, please report it.  For instructions, see:
<http://www.gnu.org/software/gdb/bugs/>.

/build/gdb-ktlO15/gdb-9.2/gdb/regcache.c:166: internal-error: int register_size(gdbarch*, int): Assertion `regnum >= 0 && regnum < gdbarch_num_cooked_regs (gdbarch)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Create a core file of GDB? (y or n) [answered Y; input not from terminal]
Aborted (core dumped)
```

</details>

[the remark this answers is not in the document]
Comment (elhimov) [reply tools/tarantool-gdb.py:2685]: Does it crash with the only fiber (102)? Would you, please, check with 101 for example?

[the remark this answers is not in the document]
Comment (locker) [reply tools/tarantool-gdb.py:2685]: `tt-fiber` crashes for every fiber shown by `info tt-fiber`.

<details>

```
(gdb) tt-fiber 101
/build/gdb-ktlO15/gdb-9.2/gdb/regcache.c:166: internal-error: int register_size(gdbarch*, int): Assertion `regnum >= 0 && regnum < gdbarch_num_cooked_regs (gdbarch)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) [answered Y; input not from terminal]

This is a bug, please report it.  For instructions, see:
<http://www.gnu.org/software/gdb/bugs/>.

/build/gdb-ktlO15/gdb-9.2/gdb/regcache.c:166: internal-error: int register_size(gdbarch*, int): Assertion `regnum >= 0 && regnum < gdbarch_num_cooked_regs (gdbarch)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Create a core file of GDB? (y or n) [answered Y; input not from terminal]
Aborted (core dumped)
```

</details>

[the remark this answers is not in the document]
Comment (elhimov) [reply tools/tarantool-gdb.py]: Crash is fixed, thanks for help.

**Review by locker — APPROVED** [review]

**Pull request merged**